docs/sources/clients/promtail/configuration.md

```diff
@@ -879,6 +885,70 @@ labels:
 [use_incoming_timestamp: <bool> | default = false]
 ```
 
+### kafka
+
+The `kafka` block configures Promtail to scrape logs from [Kafka](https://kafka.apache.org/) using a group consumer.
+
+The `brokers` should list available brokers to communicate with the Kafka cluster. Use multiple brokers when you want to increase availability.
+
+The `topics` is the list of topics Promtail will subscribe to. If a topic start with `^` then a regular expression ([RE2](https://github.com/google/re2/wiki/Syntax)) is used to match topics.
+For instance `^promtail-.*` will match the topic `promtail-dev` and `promtail-prod`. Topics are refreshed every 30 seconds, so if a new topic matches, it will be automatically added without requiring a Promtail restart.
+
+The `group_id` defined the unique consumer group id to use for consuming logs. Each log record published to a topic is delivered to one consumer instance within each subscribing consumer group.
+
+- If all promtail instances have the same consumer group, then the records will effectively be load balanced over the promtail instances.
+- If all promtail instances have different consumer groups, then each record will be broadcast to all promtail instances.
+
+The `group_id` is useful if you want to effectively send the data to multiple loki instance and/or other sinks.
+
+The `assignor` configuration allow you to select the rebalancing strategy to use for the consumer group.
+Rebalancing is the process where a group of consumer instances (belonging to the same group) co-ordinate to own a mutually exclusive set of partitions of topics that the group is subscribed to.
+
+- `range` the default, assigns partitions as ranges to consumer group members.
+- `sticky` assigns partitions to members with an attempt to preserve earlier assignments
+- `roundrobin` assigns partitions to members in alternating order.
+
+The `version` allows to select the kafka version required to connect to the cluster.(default to `2.2.1`)
+
+By default timestamps are assigned by Promtail when the message is read, if you want to keep the actual message timestamp from Kafka you can set the `use_incoming_timestamp` to true.
```

Comment:
With out of order enabled, perhaps we should make `use_incoming_timestamp` default to true. WDYT?

Reply:
Yaml 🤦 can't default to true and there's similar other configuration doing this I didn't wanted to differ from them

Reply:
ughh, ok :+1: